dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/BlockingChannelAdapter.java

```diff
@@ -52,42 +57,66 @@ private BlockingChannelAdapter() {}
     private static final class BlockingChannelAdapterChannel implements Channel {
 
         private final BlockingChannel delegate;
-        private final ListeningExecutorService executor;
+        private final ExecutorService executor;
 
-        BlockingChannelAdapterChannel(BlockingChannel delegate, ListeningExecutorService executor) {
+        BlockingChannelAdapterChannel(BlockingChannel delegate, ExecutorService executor) {
             this.delegate = delegate;
             this.executor = executor;
         }
 
         @Override
         public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
-            return executor.submit(new BlockingChannelAdapterTask(delegate, endpoint, request));
+            SettableFuture<Response> result = SettableFuture.create();
```

Comment:
s/result/settableFuture/? just makes it a tiny bit clearer which direction things are going when we're hooking up the cancel listener